pkg/storegateway/chunkscache/cache.go

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package chunkscache
+
+import (
+	"context"
+
+	gotkit_log "github.com/go-kit/log"
+	"github.com/oklog/ulid"
+	"github.com/opentracing/opentracing-go/log"
+	"github.com/prometheus/prometheus/tsdb/chunks"
+
+	"github.com/grafana/mimir/pkg/util/spanlogger"
+)
+
+type Range struct {
+	BlockID   ulid.ULID
+	Start     chunks.ChunkRef
+	NumChunks int
+}
+
+type ChunksCache interface {
+	FetchMultiChunks(ctx context.Context, userID string, ranges []Range) (hits map[Range][]byte)
+	StoreChunks(ctx context.Context, userID string, r Range, v []byte)
+}
+
+type TracingCache struct {
+	C ChunksCache
```

Comment:
Does `C` needs to be exported?

Reply:
otherwise we'd need a constructor for the TracingCache. This way you just use a struct literal. Do you prefer a constructor?

Reply:
See comment about the logger. If you accept that comment, then I would go with the constructor.